### PR TITLE
[improve] [broker] Trigger offload on topic load

### PR DESCRIPTION
**Pull request by hangc0276**
Fixes #xyz

Main Issue: #xyz

PIP: [PIP-348](https://github.com/apache/pulsar/pull/22650)

### Motivation

Implement of [PIP-348](https://github.com/apache/pulsar/pull/22650)

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by poorbarcode — APPROVED** [review]

**Review by horizonzy — APPROVED** [review]
LGTM.

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java [hunk 1]
```diff
@@ -395,6 +396,8 @@ public void initializeComplete() {
 
                     // May need to update the cursor position
                     newledger.maybeUpdateCursorBeforeTrimmingConsumedLedger();
+                    // May need to trigger offloading
+                    newledger.maybeOffloadInBackground(NULL_OFFLOAD_PROMISE);
```
Comment (dao-jun) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java:400]: Just a suggestion, maybe trigger offloading in BrokerService#createPersistentTopic is better?
Trigger offloading after PersistentTopic load finished:
image

Comment (hangc0276) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java:400]: Offload is a managedledger level feature and does not need to be exposed as the topic-level feature. From the feature scope perspective, we'd better keep it at the managedledger level.

Comment (dao-jun) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java:400]: It makes sense, and it seems it's OK even load topic failed

**Review by dao-jun — APPROVED** [review]

**Review by Technoboy- — APPROVED** [review]

**Comment by dao-jun** [comment]
close reopen to trigger ci workflow

**Pull request merged**